modules/azure/cosmosdb.go

```diff
@@ -0,0 +1,185 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/preview/cosmos-db/mgmt/documentdb"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// GetCosmosDBAccountClient is a helper function that will setup a CosmosDB account client.
+func GetCosmosDBAccountClient(subscriptionID string) (*documentdb.DatabaseAccountsClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
```

Comment:
Switch GetClient function to have sovereign cloud support.  doc for how to do this is [here](https://terratest.gruntwork.io/docs/community/contributing/#azure-sdk-client-factory).

Reply:
Hey Richard, I implemented this, however the tests fail when running in my tenant (the MS one), for China and German clouds (see error below)... Do you know how to get around this? Or will this work in your CI environment?

```
***"error":"invalid_resource","error_description":"AADSTS500011: The resource principal named https://management.chinacloudapi.cn/ was not found in the tenant named 72f988bf-86f1-41af-91ab-2d7cd011db47. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant.\r\nTrace ID: 355d2142-339d-4d8e-ac28-bbe39a2e2b00\r\nCorrelation ID: 84fdfe4e-c3b5-49ba-bbc8-87b7ea12ea03\r\nTimestamp: 2020-12-21 11:49:18Z","error_codes":[500011],"timestamp":"2020-12-21 11:49:18Z","trace_id":"355d2142-339d-4d8e-ac28-bbe39a2e2b00","correlation_id":"84fdfe4e-c3b5-49ba-bbc8-87b7ea12ea03","error_uri":"https://login.microsoftonline.com/error?code=500011"***
```

Here is my CI run: https://github.com/tomconte/terratest/runs/1588685618?check_suite_focus=true

Reply:
Never mind, I found what was wrong. Should be OK now.